### ISSUE TITLE
iOS Encountered error while building for device

### ISSUE THREAD

**Issue by Abdullah92h** [flutter/flutter#88680]
I tried to run my app with `flutter build ios` and I got this message `Encountered error while building for device.`

**I have tried:**
`flutter clean && \ rm ios/Podfile ios/Podfile.lock pubspec.lock && \ rm -rf ios/Pods ios/Runner.xcworkspace && flutter build iOS`

But still got the same error

**Expected results:** 

I want my app to run on the simulator

**Actual results:** 

<details>
  <summary>Logs</summary>

**flutter run**

```
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/abdullahhamadi/Development/tools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_
    handler-5.0.1+2/ios/Classes/strategies/PhonePermissionStrategy.m:43:34: error: no visible
    @interface for 'NSDictionary<NSString *,CTCarrier *>' declares the selector
    'mobileNetworkCode'
            NSString *mnc = [carrier mobileNetworkCode];
                             ~~~~~~~ ^~~~~~~~~~~~~~~~~
    /Users/abdullahhamadi/Development/tools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_
    handler-5.0.1+2/ios/Classes/strategies/PhonePermissionStrategy.m:43:19: warning: unused
    variable 'mnc' [-Wunused-variable]
            NSString *mnc = [carrier mobileNetworkCode];
                      ^
    /Users/abdullahhamadi/Development/tools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_
    handler-5.0.1+2/ios/Classes/strategies/PhonePermissionStrategy.m:47:11: error: use of
    undeclared identifier 'mnc'
        if (([mnc length] == 0) || ([mnc isEqualToString:@"65535"])) {
              ^
    /Users/abdullahhamadi/Development/tools/flutter/.pub-cache/hosted/pub.dartlang.org/permission_
    handler-5.0.1+2/ios/Classes/strategies/PhonePermissionStrategy.m:47:34: error: use of
    undeclared identifier 'mnc'
        if (([mnc length] == 0) || ([mnc isEqualToString:@"65535"])) {
                                     ^
    1 warning and 3 errors generated.
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Analyzing workspace
    note: Constructing build description
    note: Build preparation complete

Could not build the application for the simulator.
Error launching application on iPhone 12 Pro Max.
```
</details>

**Podfile**
```
# Uncomment this line to define a global platform for your project
platform :ios, '10.0'

# CocoaPods analytics sends network stats synchronously affecting flutter build latency.
ENV['COCOAPODS_DISABLE_STATS'] = 'true'

project 'Runner', {
  'Debug' => :debug,
  'Profile' => :release,
  'Release' => :release,
}

def flutter_root
  generated_xcode_build_settings_path = File.expand_path(File.join('..', 'Flutter', 'Generated.xcconfig'), __FILE__)
  unless File.exist?(generated_xcode_build_settings_path)
    raise "#{generated_xcode_build_settings_path} must exist. If you're running pod install manually, make sure flutter pub get is executed first"
  end

  File.foreach(generated_xcode_build_settings_path) do |line|
    matches = line.match(/FLUTTER_ROOT\=(.*)/)
    return matches[1].strip if matches
  end
  raise "FLUTTER_ROOT not found in #{generated_xcode_build_settings_path}. Try deleting Generated.xcconfig, then run flutter pub get"
end

require File.expand_path(File.join('packages', 'flutter_tools', 'bin', 'podhelper'), flutter_root)

flutter_ios_podfile_setup

target 'Runner' do
  use_frameworks!
  use_modular_headers!

  flutter_install_all_ios_pods File.dirname(File.realpath(__FILE__))
end

post_install do |installer|
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings['ENABLE_BITCODE'] = 'NO'
      config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
    end
    flutter_additional_ios_build_settings(target)
  end
end
```

**flutter doctor -v**

```
[✓] Flutter (Channel stable, 2.0.6, on macOS 11.5.2 20G95 darwin-x64, locale en-US)
    • Flutter version 2.0.6 at /Users/abdullahhamadi/Development/tools/flutter
    • Framework revision 1d9032c7e1 (4 months ago), 2021-04-29 17:37:58 -0700
    • Engine revision 05e680e202
    • Dart version 2.12.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/abdullahhamadi/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/abdullahhamadi/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.59.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.25.0

[✓] Connected device (2 available)
    • iPhone 12 Pro Max (mobile) • 855D4BBB-13EA-483F-9F15-D55DE7053774 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)
    • Chrome (web)               • chrome                               • web-javascript • Google
      Chrome 92.0.4515.159

• No issues found!
```

[verbose log.txt](https://github.com/flutter/flutter/files/7029282/verbose.log.txt)

**Comment by TahaTesser** [comment]
Hi @Abdullah92h 

First of all, upgrade to the latest Flutter stable channel 

Can you please upgrade to the latest  `stable` and try again
Execute the following commands to upgrade
`flutter channel stable`
`flutter upgrade --force`
`flutter doctor -v`


Please update your Podfile with this at the end and then try again 
```diff
post_install do |installer|
  installer.pods_project.targets.each do |target|
-   target.build_configurations.each do |config|
-    config.build_settings['ENABLE_BITCODE'] = 'NO'
-    config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = '10.0'
+   flutter_additional_ios_build_settings(target)
    end
end
```

Thank you

**Comment by darshankawar** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
